### PR TITLE
getopt: assert on duplicate long/short options

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Fixes https://issues.dlang.org/show_bug.cgi?id=16048

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Hah, it uncovered a bug in the DMD test-suite.

**Comment by AndrejMitrovic** [comment]
Sigh, something so simple should just work. I really dislike the template free-function mess in Phobos.

File: std/getopt.d [hunk 1]
```diff
@@ -681,6 +684,24 @@ private void getoptImpl(T...)(ref string[] args, ref configuration cfg,
...
+                excep = new GetOptException("Long option " ~ optionHelp.optLong
+                    ~ " is multiply defined");
+                return;
+            }
+
+            if (optionHelp.optShort.length &&
+                optionHelp.optShort in visitedShortOpts)
+            {
+                excep = new GetOptException("Short option " ~ optionHelp.optShort
+                    ~ " is multiply defined");
```
Comment (jmdavis) [inline std/getopt.d]: Same here.

**Comment by jmdavis** [comment]
Why isn't this done with an assertion? This is finding a programming bug, not bad user input. This is the sort of thing that `assert` is for.

**Comment by AndrejMitrovic** [comment]
> Why isn't this done with an assertion? This is finding a programming bug, not bad user input. This is the sort of thing that assert is for.

Hmm good point! I'll fix it.

**Comment by wilzbach** [comment]
> Nice! Thanks for the fix!

Imho this is a valid fix and nothing is blocking its merge, so I will go ahead and merge it, s.t. it's part of the upcoming 2.073 release.

**Comment by wilzbach** [comment]
Auto-merge toggled on

**Pull request merged**